@slack/client incompatible with @types/bluebird@2.x

### Description

If a typescript project is using `@types/bluebird@2.0.33`, it cannot use `@slack/client`.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Reproducible in:

`@slack/client` version: 4.3.1

node version: 8.9.1

OS version(s):

#### Steps to reproduce:

```bash
$ mkdir demo && cd demo
$ npm install typescript@3.0.1 @types/bluebird@2.0.33 @slack/client@4.3.1
$ touch index.ts
$ tsc --lib es2017 index.ts
node_modules/@types/p-cancelable/index.d.ts:10:11 - error TS2430: Interface 'PCancelableConstructor' incorrectly extends interface 'PromiseConstructor'.
  Types of property 'prototype' are incompatible.
    Type 'PCancelable<any>' is not assignable to type 'Promise<any>'.
      Types of property 'cancel' are incompatible.
        Type '() => void' is not assignable to type '<U>(reason?: any) => Promise<U>'.
          Type 'void' is not assignable to type 'Promise<U>'.

10 interface PCancelableConstructor extends PromiseConstructor {
             ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/p-cancelable/index.d.ts:18:15 - error TS2430: Interface 'PCancelable<T>' incorrectly extends interface 'Promise<T>'.
  Types of property 'cancel' are incompatible.
    Type '() => void' is not assignable to type '<U>(reason?: any) => Promise<U>'.
      Type 'void' is not assignable to type 'Promise<U>'.

18     interface PCancelable<T> extends Promise<T> {
                 ~~~~~~~~~~~
```
#### Expected result:

compiles

#### Actual result:

doesn't compile

I got a similar problem in my CI with latest `v4.4.0`.

```
$ mkdir demo
$ cd demo
$ npm install typescript@3.0.1 @slack/client@4.4.0
$ echo import { RTMClient } from '@slack/client'; > index.ts
$ tsc --lib es2017 index.ts
node_modules/@slack/client/dist/util.d.ts:32:56 - error TS2304: Cannot find name 'AsyncIterable'.

32 export declare function awaitAndReduce<T, U>(iterable: AsyncIterable<T>, callbackfn: (previousValue: U, currentValue: T) => U, initialValue: U): Promise<U>;
                                                          ~~~~~~~~~~~~~
```
Posting it here because it's a Typescript related problem, let me know if you prefer me creating new issue.

--- a/docs/_pages/typescript.md
+++ b/docs/_pages/typescript.md
@@ -23,7 +23,7 @@ This page helps describe how to use this package from a project that also uses T
 In your project, you'll need to make sure that certain `lib`s are referenced. Usually you would configure this inside
 a `tsconfig.json` file at the root of your project. If you haven't already created this file, you can generate one
 using `tsc --init` on the command line. Below is an example of what the `lib` should _at a minimum_ contain for a
-prject using this package.
+project using this package.
 
 For versions of TypeScript < 3.1:
 ```
@@ -59,8 +59,8 @@ The above options will help you compile down to an `"es5"` target, but if you're
 
 ### Result properties are not defined
 
-This package aims to provides complete and accurate type information for all method **inputs** (or arguments), but
-doesn't necessarily provide that depth for **outputs** or the results. This is a shortcoming we are aware of, and are
+This package aims to provide complete and accurate type information for all method **inputs** (or arguments), but
+doesn't necessarily provide that depth for **outputs** (or the results). This is a shortcoming we are aware of, and are
 evaluating the various ways we might be able to address it in issues
 [#496](https://github.com/slackapi/node-slack-sdk/issues/496) and
 [#509](https://github.com/slackapi/node-slack-sdk/issues/509). There are two recommendations for working with results.

--- a/docs/_pages/typescript.md
+++ b/docs/_pages/typescript.md
@@ -1,0 +1,118 @@
+---
+layout: page
+title: Using TypeScript
+permalink: /typescript
+order: 5
+headings:
+    - title: Configuration requirements
+    - title: Result properties are not defined
+    - title: Known issues
+---
+
+This project is written and built using [TypeScript](https://www.typescriptlang.org/), which means all of the APIs
+have type information metadata :tada:. This means that if you're using a code editor like VSCode, Atom, or many others
+that know how to read that metadata, or if you're using TypeScript in your own project, you'll benefit from improved
+documentation as your write code, early detection of errors, easier refactoring, and more.
+
+This page helps describe how to use this package from a project that also uses TypeScript.
+
+---
+
+### Configuration requirements
+
+In your project, you'll need to make sure that certain `lib`s are referenced. Usually you would configure this inside
+a `tsconfig.json` file at the root of your project. If you haven't already created this file, you can generate one
+using `tsc --init` on the command line. Below is an example of what the `lib` should _at a minimum_ contain for a
+prject using this package.
+
+For versions of TypeScript < 3.1:
+```
+{
+  "compilerOptions": {
+    "lib": [
+      "es5",
+      "es2015",
+      "es2016.array.include",
+      "esnext.asynciterable"
+    ]
+  }
+}
+```
+
+For versions of TypeScript >= 3.1:
+```
+{
+  "compilerOptions": {
+    "lib": [
+      "es5",
+      "es2015",
+      "es2016.array.include",
+      "es2018.asynciterable"
+    ]
+  }
+}
+```
+
+The above options will help you compile down to an `"es5"` target, but if you're compiling to a target newer than that, you won't need to explicitly include the older `"lib"` values. Also, if you are willing to include a larger set of features from a specific language version, you don't need to include each minor feature on its own (e.g. `"es2016"` would cover the need for `"es2016.array.include"`).
+
+---
+
+### Result properties are not defined
+
+This package aims to provides complete and accurate type information for all method **inputs** (or arguments), but
+doesn't necessarily provide that depth for **outputs** or the results. This is a shortcoming we are aware of, and are
+evaluating the various ways we might be able to address it in issues
+[#496](https://github.com/slackapi/node-slack-sdk/issues/496) and
+[#509](https://github.com/slackapi/node-slack-sdk/issues/509). There are two recommendations for working with results.
+
+The first recommendation is to create an type that describes the result you expect based on the information available
+in the [Web API method documentation](https://api.slack.com/methods). This is the preferred way, since you'll still
+have rich typing available as the value is used around your codebase.
+
+```typescript
+import { WebClient, WebAPICallResult } from '@slack/client';
+
+const web = new WebClient(process.env.SLACK_TOKEN);
+
+interface ChatPostMessageResult extends WebAPICallResult {
+  channel: string;
+  ts: string;
+  message: {
+    text: string;
+  }
+}
+
+web.chat.postMessage({ text: 'Hello world', channel: 'C012345' })
+  .then((rawResult) => {
+    const res = rawResult as ChatPostMessageResult;
+    console.log(`A message was posed to conversation ${res.channel} with id ${res.ts} which contains the message ${res.message}`);
+  })
+  .catch(console.error);
+```
+
+Notice that we didn't need to create a complete definition for the result, just something that is accurate and
+describes the properties which this app needs.
+
+The second recommendation is less preferred, but its a useful hack for moving quickly. You can cast the result to `any`.
+
+```typescript
+import { WebClient, WebAPICallResult } from '@slack/client';
+
+const web = new WebClient(process.env.SLACK_TOKEN);
+
+web.chat.postMessage({ text: 'Hello world', channel: 'C012345' })
+  .then((rawResult) => {
+    const res = rawResult as any;
+    console.log(`A message was posed to conversation ${res.channel} with id ${res.ts} which contains the message ${res.message}`);
+  })
+  .catch(console.error);
+```
+
+---
+
+### Known issues
+
+Since type information metadata is many times a crowdsourced resource provided by `@types/*` packages from
+DefinitelyTyped, certain packages might conflict with one another. One such case is the `@types/bluebird@1.x` package.
+If you need to use `bluebird`, we recommend updating to a newer version (at least v2). Details about this conflict can
+be found in issue [#612](https://github.com/slackapi/node-slack-sdk/issues/612).

